### DISCUSSION TITLE
"tsc" command errors

### DISCUSSION THREAD

**Discussion by lucksp** [vercel/next.js#23026] [Help]
I am setting up Typescript to an existing JS project. Always fun ;).   As part of my commit process, with `lint-staged` I would like to run `tsc` command as part of the pre-commit hook.  However, the TSC complier is displaying some errors that are not showing up within the VSCode environment.  I would like to think this is because of the special sauce of NextJS...

When i run: `"type-check": "tsc --noEmit --skipLibCheck"` the following errors are displayed:

> error TS1259: Module '".../node_modules/@types/react/index"' can only be default-imported using the 'esModuleInterop' flag

> error TS6142: Module '../components/Reservation/CartBillingMethod' was resolved to 'components/Reservation/CartBillingMethod.jsx', but '--jsx' is not set.

> error TS17004: Cannot use JSX unless the '--jsx' flag is provided.

My tsconfig should be OK:

```
{
  "compilerOptions": {
    "target": "es5",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": true,
    "allowSyntheticDefaultImports": true,
    "skipLibCheck": true,
    "strict": false,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve", // Next always sets to preserve
    "baseUrl": ".",
    "paths": {
      "@/*": [
        "./*"
      ]
    },
    "types": []
  },
  "include": [
    "next-env.d.ts",
    "**/*.ts",
    "**/*.tsx"
  ],
  "exclude": [
    "node_modules",
    "cypress",
    ".next",
    ".nyc_output"
  ]
}
```
 

Is this a feature that cannot be run?

**Comment by DollarAkshay** [comment]
Did you find any solution to this ?
